js/Repeater/RepeaterComponent.js

```diff
@@ -0,0 +1,513 @@
+import $ from 'jquery';
+import _ from 'lodash';
+
+class Repeater {
+    /**
+     * Repeater
+     */
+    constructor (window) {
+        this.repeaterEntries = 0;
+        this.savedEntries = 0;
+        this.window = window;
+
+        // read-only
+        this.repeaterAttributes = {
+            name: 'data-repeater-name',
+            newGroup: 'data-repeater-new-group',
+            addGroup: 'data-repeater-add-group',
+            editGroup: 'data-repeater-edit-group',
+            deleteGroup: 'data-repeater-delete-group',
+            forName: 'data-repeater-for-name',
+            entryId: 'data-repeater-entry-id',
+            previewId: 'data-repeater-preview-id',
+            editId: 'data-repeater-edit-id',
+            saveGroup: 'data-repeater-save-group',
+            cancelSave: 'data-repeater-cancel-save',
+            updateId: 'data-repeater-update-id',
+            savedDataId: 'data-repeater-saved-data-id',
+            addNewGroupText: 'data-repeater-add-new-group-text',
+            addAnotherGroupText: 'data-repeater-add-another-group-text',
+            maxEntries: 'data-repeater-max-entries',
+            previewUi: 'data-repeater-preview-ui'
+        };
+
+        // Merge these with options
+        // TODO: These will probably be read-only
+        this.repeaterQueries = {
+            newGroup: { query: `[${this.repeaterAttributes.newGroup}]` },
+            addGroup: { query: `[${this.repeaterAttributes.addGroup}]` },
+            saveGroup: { query: `[${this.repeaterAttributes.saveGroup}]` },
+            cancelSave: { query: `[${this.repeaterAttributes.cancelSave}]` },
+            savedEntryRoot: { query: '.repeater__saved-data' },
+            previewDataRoot: { query: '.repeater__preview-data' },
+            entryData: { query: '.repeater__entry-data' },
+            previewHeadings: { query: `[${this.repeaterAttributes.forName}]` },
+            placeholder: { query: '.repeater__empty-placeholder' },
+            editGroup: { query: `[${this.repeaterAttributes.editGroup}]` },
+            deleteGroup: { query: `[${this.repeaterAttributes.deleteGroup}]` },
+            preview: { query: `[${this.repeaterAttributes.previewId}]` },
+            previewUi: { query: `[${this.repeaterAttributes.previewUi}]` }
+        };
+
+        // Preview UI HTML
+        this.previewUiHTML = `
+            <td class="table__td table__td--align-right table-row__actions">                
```

Comment:
`table__td` and `table__td--align-right` are not required, they're legacy styles, most of the table styles are currently on element selectors so the top level table _should_ just need `.table`